mongo/mongo.go

```diff
@@ -27,37 +31,55 @@ const imagesLockCol = "images_locks"
 type Mongo struct {
 	Collection   string
 	Database     string
-	Session      *mgo.Session
+	Connection   *dpMongoDriver.MongoConnection
 	URI          string
+	Username     string
+	Password     string
 	client       *dpMongoHealth.Client
 	healthClient *dpMongoHealth.CheckMongoClient
 	lockClient   *dpMongoLock.Lock
+	IsSSL        bool
 }
 
 // Init creates a new mgo.Session with a strong consistency and a write mode of "majority".
-func (m *Mongo) Init(ctx context.Context) (err error) {
-	if m.Session != nil {
-		return errors.New("session already exists")
+func (m *Mongo) getConnectionConfig(shouldEnableReadConcern, shouldEnableWriteConcern bool) *dpMongoDriver.MongoConnectionConfig {
+	return &dpMongoDriver.MongoConnectionConfig{
+		IsSSL:                   m.IsSSL,
+		ConnectTimeoutInSeconds: connectTimeoutInSeconds,
+		QueryTimeoutInSeconds:   queryTimeoutInSeconds,
+
+		Username:                      m.Username,
+		Password:                      m.Password,
+		ClusterEndpoint:               m.URI,
+		Database:                      m.Database,
+		Collection:                    m.Collection,
+		IsWriteConcernMajorityEnabled: shouldEnableWriteConcern,
+		IsStrongReadConcernEnabled:    shouldEnableReadConcern,
 	}
+}
 
-	// Create session
-	if m.Session, err = mgo.Dial(m.URI); err != nil {
+// Init creates a new mgo.Session with a strong consistency and a write mode of "majority".
+func (m *Mongo) Init(ctx context.Context, shouldEnableReadConcern, shouldEnableWriteConcern bool) (err error) {
+	if m.Connection != nil {
+		return errors.New("Datastor Connection already exists")
+	}
+	mongoConnection, err := dpMongoDriver.Open(m.getConnectionConfig(shouldEnableReadConcern, shouldEnableWriteConcern))
+	if err != nil {
 		return err
 	}
-	m.Session.EnsureSafe(&mgo.Safe{WMode: "majority"})
-	m.Session.SetMode(mgo.Strong, true)
+	m.Connection = mongoConnection
 
 	databaseCollectionBuilder := make(map[dpMongoHealth.Database][]dpMongoHealth.Collection)
 	databaseCollectionBuilder[(dpMongoHealth.Database)(m.Database)] = []dpMongoHealth.Collection{(dpMongoHealth.Collection)(m.Collection), (dpMongoHealth.Collection)(imagesLockCol)}
-	// Create client and healthclient from session
-	m.client = dpMongoHealth.NewClientWithCollections(m.Session, databaseCollectionBuilder)
+	// Create client and health-client from session
+	m.client = dpMongoHealth.NewClientWithCollections(m.Connection, databaseCollectionBuilder)
 	m.healthClient = &dpMongoHealth.CheckMongoClient{
 		Client:      *m.client,
 		Healthcheck: m.client.Healthcheck,
 	}
 
 	// Create MongoDB lock client, which also starts the purger loop
-	m.lockClient = dpMongoLock.New(ctx, m.Session, m.Database, imagesCol)
+	m.lockClient = dpMongoLock.New(ctx, m.Connection, imagesLockCol)
```

Comment:
This is now using `imagesLockCol` instead of `imagesCol` - is that intended? 

did the lock library used to append the `_locks` onto the collection name or something? I am trying to understand why this needed to change.

Reply:
@CarlHembrough  Earlier the locks library was creating the `_locks` suffix. Now this was changed to the service using it passing the locked collection name.